pkg/controller/utils/elasticsearch.go

```diff
@@ -149,17 +151,17 @@ func NewElasticClient(client client.Client, ctx context.Context, elasticHTTPSEnd
 	return &esClient{client: esCli}, err
 }
 
-func formatName(name string, clusterName, tenantID string, management bool) string {
+func formatName(name string, clusterID, tenantID string, management bool) string {
 	if tenantID != "" {
-		return fmt.Sprintf("%s_%s", name, tenantID)
+		return fmt.Sprintf("%s_%s_%s", name, clusterID, tenantID)
 	}
 
 	var formattedName string
 
 	if management {
 		formattedName = string(name)
 	} else {
```

Comment:
TBH, I am not really sure why this code is the way that it is... In practice this code is only executed today for multi-tenancy so I think all of this below the first tenantID branch is dead code anyway. Might just be worth deleting it to remove confusion...

Reply:
Yeah good catch, I'll remove and updated tests accordingly.